Sources/XCTest/Public/XCTestCase.TearDownBlocksState.swift

```diff
@@ -0,0 +1,48 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//
+//  XCTestCase.ClosureType.swift
+//  Extension on XCTestCase which encapsulates ClosureType
+//
+
+///  XCTestCase.TeardownBlocksState
+///  A class which encapsulates teardown blocks which are registered via the `addTearDownBlock(_block:)` method.
+///  Supports async and sync throwing methods
+
+extension XCTestCase {
+    final class TeardownBlocksState {
+        
+        private var wasFinalized = false
+        private var blocks: [() throws -> Void] = []
+        
+        @available(macOS 12, *)
+        func append(_ block: @escaping () async throws -> Void) {
```

Comment:
Similar my comment on the API which calls this, I think this closure param should be marked `@Sendable`
```suggestion
        func append(_ block: @Sendable @escaping () async throws -> Void) {
```